lib/eventPosition.ts

```diff
@@ -0,0 +1,177 @@
+   * if the position is just created from an enqueued time or an offset.
+   */
+  sequenceNumber?: number;
+
+  /**
+   * @property {string} [customFilter] The custom filter expression that needs to be set on the receiver.
+   */
+  customFilter?: string;
+}
+
+/**
+ * Represents options can be set during the creation of a event hub receiver.
+ * Defines a position of an @link~EventData in the event hub partition.
+ * @class EventPosition
+ */
+export default class EventPosition {
+
+  static readonly startOfStream: string = "-1";
+
+  static readonly endOfStream: string = "@latest";
+
+  /**
+   * @property {string} [offset] The offset of the event at the position. It can be undefined
+   * if the position is just created from a sequence number or an enqueued time.
+   */
+  offset?: string;
+  /**
+   * @property {boolean} isInclusive Indicates if the current event at the specified offset is
+   * included or not. It is only applicable if offset is set. Default value: false.
+   */
+  isInclusive: boolean = false;
+  /**
+   * @property {Date|number} [enqueuedTime] The enqueued time of the event at the position. It can be undefined
+   * if the position is just created from a sequence number or an offset.
+   */
+  enqueuedTime?: Date | number;
+
+  /**
+   * @property {number} [sequenceNumber] The sequence number of the event at the position. It can be undefined
+   * if the position is just created from an enqueued time or an offset.
+   */
+  sequenceNumber?: number;
+
+  /**
+   * @property {string} [customFilter] The custom filter expression that needs to be set on the receiver.
+   */
+  customFilter?: string;
+
+  constructor(options?: EventPositionOptions) {
+    if (options) {
+      this.offset = options.offset;
+      this.enqueuedTime = options.enqueuedTime;
+      this.sequenceNumber = options.sequenceNumber;
+      this.isInclusive = options.isInclusive || false;
+      this.customFilter = options.customFilter;
+    }
+  }
+
+  /**
+   * Gets the expression (filter clause) that needs to be set on the source.
+   * @return {string} filterExpression
+   */
+  getExpression(): string {
+    let result;
+    // order of preference
+    if (this.offset) {
+      result = this.isInclusive ?
+        `${Constants.offsetAnnotation} >= '${this.offset}'` :
+        `${Constants.offsetAnnotation} > '${this.offset}'`;
+    } else if (this.sequenceNumber) {
+      result = this.isInclusive ?
+        `${Constants.sequenceNumberAnnotation} >= '${this.sequenceNumber}'` :
+        `${Constants.sequenceNumberAnnotation} > '${this.sequenceNumber}'`;
+    } else if (this.enqueuedTime) {
+      let time = (this.enqueuedTime instanceof Date) ? this.enqueuedTime.getTime() : this.enqueuedTime;
+      result = `${Constants.enqueuedTimeAnnotation} > '${time}'`;
+    } else if (this.customFilter) {
+      result = this.customFilter;
+    }
+
+    if (!result) {
+      throw new Errors.ArgumentError("No starting position was set in the EventPosition.");
```

Comment:
does it make sense to use an argument error for your parameter type validations as well?